itwin-platform-access/imodels-access-common/src/ErrorAdapter.ts

```diff
@@ -32,11 +31,21 @@ export class ErrorAdapter {
     if (error.code === IModelsErrorCode.InvalidIModelsRequest)
       return ErrorAdapter.adaptInvalidRequestErrorIfPossible(error);
 
-    let errorNumber = ErrorAdapter.tryMapGenericErrorCode(error.code, operationName);
-    if (!errorNumber)
-      errorNumber = ErrorAdapter.mapErrorCode(error.code);
-
-    return new IModelError(errorNumber, error.message);
+    let errorCode = ErrorAdapter.tryMapGenericErrorCode(error.code, operationName);
+    if (!errorCode)
+      errorCode = ErrorAdapter.mapErrorCode(error.code);
+
+    if ("conflictingLocks" in error)
+      return ITwinError.create<ITwinError & { conflictingLocks?: unknown }>({
```

Comment:
Can you define an export an interface for this error type?

Reply:
See how iTwin Errors are [used in itwinjs-core](https://github.com/iTwin/itwinjs-core/blob/13f8570ae530ddcf3cb29b29ca0c18e207a714c6/core/common/src/ITwinCoreErrors.ts#L38).

Would be nice to have something like `IModelsErrors.isError(err, IModelsErrorCode.ConflictWithAnotherUser): err is ConflictingLocksError`.